nodewatcher/modules/frontend/ip_space/static/js/draw_cidrs.js

```diff
@@ -0,0 +1,297 @@
+                        self.draw(data.results[j].network + "/" + data.results[j].prefix_length, data.results[j].description);
+                    }
+                    if (self.loadedPrefixes == 32) {
+                        jQuery('rect').tipsy({
+                            gravity: 'w',
+                            html: true,
+                            title: function() {
+                                return this.id + '<br>Number of hosts: ' + $(this).attr('n') + '<br>Network name: ' + $(this).attr('d');
+                            }
+                        });
+                        self.setTopNodes(self.calcTopNodes(self.data));
+                    }
+                }
+
+            });
+    }
+
+    load() {
+        for (var i = 0; i < 33; i++) {
+            this.loadPrefix(i, 'api/v2/pool/ip/?format=json&prefix_length=' + i);
+        }
+    }
+
+    cidrToRange(cidr) {
+       var range = [2];
+       cidr = cidr.split('/');
+       var cidr_1 = parseInt(cidr[1])
+       range[0] = (this.ip2num(cidr[0])) & ((-1 << (32 - cidr_1)));
+       var start = this.ip2num(range[0])
+       range[1] = range[0] + Math.pow(2, (32 - cidr_1)) - 1;
+       return [range[0]-this.offset, range[1]-this.offset];
+    }
+
+    calcTopNodes() {
+        var res = new Array();
+        for (var i = 0; i < this.data.length; i++) {
+            for (var j = 0; j < this.data[i].length; j++) {
+                var node = this.data[i][j];
+                if (node['@id'] == node.top_level['@id']) {
+                    res.push(node);
+                }
+            }
+        }
+        return res;
+    }
+
+    getMaxOfArray(numArray) {
+      return Math.max.apply(null, numArray);
+    }
+
+    getMinOfArray(numArray) {
+      return Math.min.apply(null, numArray);
+    }
+
+    setTopNodes(data) {
+        var self = this;
+        var xs = [];
+        var ys = [];
+        data.forEach(function(node) {
+            xs.push(d2xy(self.cidrToRange(node.network + '/' + node.prefix_length)[0]).x);
+            ys.push(d2xy(self.cidrToRange(node.network + '/' + node.prefix_length)[0]).y);
+            xs.push(d2xy(self.cidrToRange(node.network + '/' + node.prefix_length)[1]).x);
+            ys.push(d2xy(self.cidrToRange(node.network + '/' + node.prefix_length)[1]).y);
+        });
+
+        var min_x = self.getMinOfArray(xs);
+        var min_y = self.getMinOfArray(ys);
+
+        var max_x = self.getMaxOfArray(xs);
+        var max_y = self.getMaxOfArray(ys);
+
+
+        var start_num = xy2d(min_x, min_y);
+        var stop_num = xy2d(max_x, max_y);
+
+        var ips_needed = stop_num - start_num;
+        var mask = 0;
+        while(Math.pow(2, mask) < ips_needed){
+            mask++;
+        }
+        mask = 32-mask;
+        var start_ip = this.num2ip(this.offset + start_num);
+
+        $('#topnodes').html('<li id="cidr" cidr="10.0.0.0/8"><a>10.0.0.0/8 (Zoom out)</a></li>');
```

Comment:
Why is `10.0.0.0/8` here? There is no zoom out to 10.0.0.0/8. 10.0.0.0/8 is just arbitrary. Zoom out always to smallest network which fits.

Reply:
Why is `10.0.0.0/8` still here?

Reply:
You are recreating whole topnodes?